session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -396,6 +519,42 @@ public void insertRecords(List<String> deviceIds, List<Long> times,
     }
   }
 
+  /**
+   * Insert multiple rows in asynchronous way. This method is just like jdbc executeBatch,
+   * we pack some insert request in batch and send them to server. If you want improve your
+   * performance, please see insertTablet method.
+   * <p>
+   * Each row is independent, which could have different deviceId, time, number of measurements
+   *
+   * @see Session#insertTablet(Tablet)
+   * @param timeout asynchronous call timeout in millisecond
+   * @param callback user provided failure callback, set to null if user does not specify.
+   */
+  public CompletableFuture<Integer> asyncInsertRecords(List<String> deviceIds, List<Long> times,
+      List<List<String>> measurementsList, List<List<String>> valuesList, long timeout,
+      FiveInputConsumer<List<String>, List<Long>, List<List<String>>, List<List<String>>, Throwable> callback) {
+    CompletableFuture<Integer> asyncRun = CompletableFuture.supplyAsync(() -> {
+      try {
+        insertRecords(deviceIds, times, measurementsList, valuesList);
+      } catch (IoTDBConnectionException | StatementExecutionException e) {
+        throw new RuntimeException(e);
+      }
+      return 0;
+    }, asyncThreadPool.getThreadPool());
+
+    return asyncRun
+        .applyToEitherAsync(orTimeout(timeout, TimeUnit.MILLISECONDS), this::successHandler)
+        .exceptionally(e -> {
+          if (callback == null) {
+            logger.error("Error occurred when inserting records, device ID: {}, time: {}.",
```

Comment:
Maybe we should print deviceID list and time list. Or you can print the length of the list, more info means easier to debug :)